include/nbl/builtin/hlsl/cpp_compat/impl/intrinsics_impl.hlsl

```diff
@@ -3,428 +3,573 @@
 {
-	static int32_t findLSB(NBL_CONST_REF_ARG(uint64_t) val)
+	static FloatingPointLikeVectorial __call(NBL_CONST_REF_ARG(FloatingPointLikeVectorial) lhs, NBL_CONST_REF_ARG(FloatingPointLikeVectorial) rhs)
 	{
 #ifdef __HLSL_VERSION
-		const uint32_t lowBits = uint32_t(val);
-		const int32_t lowLsb = find_lsb_helper<uint32_t>::findLSB(lowBits);
+		if (hlsl::is_vector_v<FloatingPointLikeVectorial>)
+			return spirv::cross(lhs, rhs);
+#else
+		using traits = hlsl::vector_traits<FloatingPointLikeVectorial>;
+		array_get<FloatingPointLikeVectorial, typename traits::scalar_type> getter;
+		array_set<FloatingPointLikeVectorial, typename traits::scalar_type> setter;
 
-		if (lowLsb == -1)
-		{
-			const uint32_t highBits = uint32_t(val >> 32);
-			const int32_t highLsb = find_lsb_helper<uint32_t>::findLSB(highBits);
-			if (highLsb == -1)
-				return -1;
-			else
-				return 32 + highLsb;
-		}
+		FloatingPointLikeVectorial output;
+		setter(output, 0, getter(lhs, 1) * getter(rhs, 2) - getter(rhs, 1) * getter(lhs, 2));
+		setter(output, 1, getter(lhs, 2) * getter(rhs, 0) - getter(rhs, 2) * getter(lhs, 0));
+		setter(output, 2, getter(lhs, 0) * getter(rhs, 1) - getter(rhs, 0) * getter(lhs, 1));
 
-		return lowLsb;
-#else
-		return glm::findLSB(val);
+		return output;
 #endif
 	}
 };
 
-template<int N>
-struct find_lsb_helper<vector<int32_t, N> >
+template<typename Vector>
+NBL_PARTIAL_REQ_TOP(is_vector_v<Vector>)
+struct clamp_helper<Vector, typename vector_traits<Vector>::scalar_type NBL_PARTIAL_REQ_BOT(is_vector_v<Vector>) >
 {
-	static vector<int32_t, N> findLSB(NBL_CONST_REF_ARG(vector<int32_t, N>) val)
+	using return_t = Vector;
+	static return_t __call(NBL_CONST_REF_ARG(Vector) val, NBL_CONST_REF_ARG(typename vector_traits<Vector>::scalar_type) min, NBL_CONST_REF_ARG(typename vector_traits<Vector>::scalar_type) max)
 	{
-#ifdef __HLSL_VERSION
-		return spirv::findILsb(val);
-#else
-		return glm::findLSB(val);
-#endif
+		using traits = hlsl::vector_traits<Vector>;
+		array_get<Vector, typename traits::scalar_type> getter;
+		array_set<return_t, typename traits::scalar_type> setter;
+
+		return_t output;
+		for (uint32_t i = 0; i < traits::Dimension; ++i)
+			setter(output, i, clamp_helper<typename traits::scalar_type, typename traits::scalar_type>::__call(getter(val, i), min, max));
+
+		return output;
 	}
 };
 
-template<int N>
-struct find_lsb_helper<vector<uint32_t, N> >
+template<typename Vector>
+NBL_PARTIAL_REQ_TOP(hlsl::is_vector_v<Vector>)
+struct bitReverse_helper<Vector NBL_PARTIAL_REQ_BOT(concepts::Vectorial<Vector>) >
 {
-	static vector<int32_t, N> findLSB(NBL_CONST_REF_ARG(vector<uint32_t, N>) val)
+	static Vector __call(NBL_CONST_REF_ARG(Vector) vec)
 	{
 #ifdef __HLSL_VERSION
-		return spirv::findILsb(val);
+		return spirv::bitReverse(vec);
 #else
```

Comment:
why have a `__HLSL_VErsioN` clause? the partial spec detecting callability of the SPIR-V intrinsic above should have caught it